### PR TITLE
test(e2e): unique names for created objects by default

### PR DESCRIPTION
**Pull request by ozyx**
Contributes towards #5513  

### Describe your changes:

- Updates `createDomainObjectWithDefaults()` appAction to generate a unique name for the object unless otherwise specified.
- Updates e2e tests to reference `object.name` instead of string literals
- Updates our `page` fixture to attach `testNotes` which includes information about the currently running test and its project configuration.
- Updates `createDomainObjectWithDefaults()` appAction to fill the created domain object's "Notes" section with our generated `testNotes`.
- Implements [ARIA: menu role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/roles/menu_role) for Menus and SuperMenus
- Updates plain `text=` selectors to be more deterministic
- Updates menu item selectors to use the new menu aria-role

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

**Review by unlikelyzero — APPROVED** [review]

File: e2e/appActions.js [hunk 1]
```diff
@@ -70,11 +75,14 @@ async function createDomainObjectWithDefaults(page, { type, name, parent = 'mine
...
-        await nameInput.fill("");
-        await nameInput.fill(name);
-    }
+    const nameInput = page.locator('form[name="mctForm"] .first input[type="text"]');
+    await nameInput.fill("");
+    await nameInput.fill(name);
+
+    // Fill the "Notes" section with information about the
+    // currently running test and its project.
+    const notesInput = page.locator('form[name="mctForm"] textarea[type="text"] >> nth=0');
```
Comment (unlikelyzero) [inline e2e/appActions.js:84]: if you have some time, can you figure out how to remove the nth locator and use something like "notes area"

File: e2e/baseFixtures.js [hunk 2]
```diff
@@ -138,11 +138,19 @@ exports.test = base.test.extend({
      * Extends the base page class to enable console log error detection.
      * @see {@link https://github.com/microsoft/playwright/discussions/11690 Github Discussion}
      */
-    page: async ({ page, failOnConsoleError }, use) => {
+    page: async ({ page, failOnConsoleError }, use, testInfo) => {
```
Comment (unlikelyzero) [inline e2e/baseFixtures.js:141]: We should move this into appActions. The baseFixtures fixture is designed to disappear as native functionality is added to the `@playwright/test` or plugin ecosystem.

In this case, the testnotes function is specific to open mct

File: e2e/helper/notebookUtils.js [hunk 3]
```diff
@@ -38,24 +40,17 @@ async function enterTextEntry(page, text) {
...
-    // Click text=Open MCT My Items >> span >> nth=3
-    await page.locator(`text=Open MCT ${myItemsFolderName} >> span`).nth(3).click();
-    // Click text=Unnamed CUSTOM_NAME
-    await Promise.all([
-        page.waitForNavigation(),
-        page.locator('text=Unnamed CUSTOM_NAME').click()
-    ]);
-
-    await page.dragAndDrop('text=UNNAMED SINE WAVE GENERATOR', NOTEBOOK_DROP_AREA);
+async function dragAndDropEmbed(page, notebookObject) {
```
Comment (unlikelyzero) [inline e2e/helper/notebookUtils.js:43]: wow

File: e2e/tests/framework/generateVisualTestData.e2e.spec.js [hunk 4]
```diff
@@ -43,22 +43,22 @@ test('Generate Visual Test Data @localStorage', async ({ page, context }) => {
     await page.locator('button:has-text("Create")').click();
 
     // add sine wave generator with defaults
-    await page.locator('li:has-text("Sine Wave Generator")').click();
+    await page.locator('li[role="menuitem"]:has-text("Sine Wave Generator")').click();
```
Comment (unlikelyzero) [inline e2e/tests/framework/generateVisualTestData.e2e.spec.js:46]: nice

File: e2e/tests/functional/search.e2e.spec.js [hunk 5]
```diff
@@ -197,75 +197,56 @@ async function waitForSearchCompletion(page) {
...
-
-    await page.locator('button:has-text("Create")').click();
-    await page.locator('li:has-text("Folder") >> nth=2').click();
-    await Promise.all([
-        page.waitForNavigation(),
-        await page.locator('text=Properties Title Notes >> input[type="text"]').fill('Blue Folder'),
-        await page.locator('form[name="mctForm"] >> text=Red Folder').click(),
-        page.locator('button:has-text("OK")').click()
-    ]);
+    const redFolder = await createDomainObjectWithDefaults(page, {
```
Comment (unlikelyzero) [inline e2e/tests/functional/search.e2e.spec.js:204]: nice

Comment (unlikelyzero) [inline e2e/tests/functional/search.e2e.spec.js:204]: nice

**Review by unlikelyzero — CHANGES_REQUESTED** [review]
Everything looks amazing, but we should move page.testNotes out of baseFixtures and into appActions. I know this won't be a clean implementation.

baseFixtures should only contain functionality which extends the base capabilities of playwright with the intent to remove our custom functions over time

**Comment by ozyx** [comment]
> baseFixtures should only contain functionality which extends the base capabilities of playwright with the intent to remove our custom functions over time

100% agree. Maybe we can move this into `pluginFixtures` instead, so we won't need to modify the function signature. I'll play around with it a bit and see what makes sense. I'm trying to avoid having to pull in the `testInfo` fixture and pass it along to `createDomainObjectWithDefaults` in every single test that creates an object.

File: src/api/menu/components/Menu.vue [hunk 6]
```diff
@@ -3,39 +3,52 @@
...
+                role="group"
             >
-                No actions defined.
-            </li>
-        </template>
+                <li
+                    v-for="action in actionGroups"
+                    :key="action.name"
+                    role="menuitem"
+                    :class="[action.cssClass, action.isDisabled ? 'disabled' : '']"
```
Comment (unlikelyzero) [inline src/api/menu/components/Menu.vue:21]: i don't know enough about this change to review this. You should quickly ping a css wiz

**Review by unlikelyzero — APPROVED** [review]

**Pull request merged**